### ISSUE TITLE
[luci/pass] Revise `--convert_nchw_to_nhwc` to handle diffusion model

### ISSUE THREAD

**Issue by jinevening** [Samsung/ONE#13478]
## What

Let's update `convert_nchw_to_nhwc` optimization to properly handle diffusion models.

## Why

**TL;DR** Diffusion model is composed of both CNN operators (convolution) and attention, which breaks the assumption of the current `convert_nchw_to_nhwc` optimization.

Below is a little bit more details.
- `convert_nchw_to_nhwc` was introduced to remove redundant transpose Ops generated from layout difference between onnx and circle. The layout difference mainly appears in CNN operators (ex: Conv2D, TransposeConv). 
- We could remove redundant transpose Ops by converting data layout of all operators except the CNN operators (Conv2D, TranseposeConv) from NCHW to NHWC (this is what `convert_nhwc_to_nhwc` does). `convert_nhwc_to_nhwc` generates a lot of transpose Ops which are eventually removed by other optimization passes. **This optimization is based on the assumption that layout of all operators in the original model (onnx) is NCHW.**
- The problem is that diffusion model is composed of both CNN operators and attention; see ResNet and Attention blocks in the figure.

![](https://jalammar.github.io/images/stable-diffusion/unet-with-text-steps-v2.png)
(Figure from https://jalammar.github.io/illustrated-stable-diffusion/)

- For such a model, we should only convert layout of ResNet block (which contains CNN operators), because Attention block does not have NCHW layout (neither NHWC). If we convert layout of Attention blocks, redundant transpose operators will remain. 

## How

Approach 1
- Detect non-CNN blocks (ex: Attention) and suppress applying `convert_nchw_to_nhwc` to them.
- Need to define a method to detect non-CNN blocks.
- This can keep the current implementation as much as possible.

Approach 2
- It would be possible to update `convert_nchw_to_nhwc` in the way that transpose Ops are propagated forward (or backward) as much as possible, anticipating that the transpose Ops are removed in the end.
- Need strategy for propagation (ex: forward/backward, multi inputs/outputs).
- This requires to rewrite most of `convert_nchw_to_nhwc`.

**Comment by jinevening** [comment]
Another problem of the previous way is misuse of this option, which happens quite often. For example, if `convert_nchw_to_nhwc` is turned on for transformer models, redundant transpose Ops are inserted, so performance is severely degraded.
